server/config.go

```diff
@@ -332,12 +336,14 @@ func (c *ScheduleConfig) adjust() {
 
 // scheduleOption is a wrapper to access the configuration safely.
 type scheduleOption struct {
-	v atomic.Value
+	v           atomic.Value
+	constraints *Constraints
 }
 
-func newScheduleOption(cfg *ScheduleConfig) *scheduleOption {
+func newScheduleOption(cfg *Config) *scheduleOption {
 	o := &scheduleOption{}
-	o.store(cfg)
+	o.store(&cfg.ScheduleCfg)
+	o.constraints = cfg.constraints
```

Comment:
Constraints are read only and will not be changed dynamically, so we don't need to protect it.